app/src/sharedTest/java/org/oppia/app/settings/profile/ProfileEditActivityTest.kt

```diff
@@ -0,0 +1,193 @@
+    ActivityScenario.launch<ProfileEditActivity>(ProfileEditActivity.createProfileEditActivity(context, 0)).use {
+      onView(withId(R.id.action_bar)).check(matches(hasDescendant(withText("Sean"))))
+      onView(withId(R.id.profile_edit_name)).check(matches(withText("Sean")))
+      onView(withId(R.id.profile_edit_allow_download_container)).check(matches(not(isDisplayed())))
+      onView(withId(R.id.profile_delete_button)).check(matches(not(isDisplayed())))
+    }
+  }
+
+  @Test
+  fun testProfileEditActivity_startActivityWithUserProfile_checkUserInfoIsDisplayed() {
+    ActivityScenario.launch<ProfileEditActivity>(ProfileEditActivity.createProfileEditActivity(context, 1)).use {
+      onView(withId(R.id.action_bar)).check(matches(hasDescendant(withText("Ben"))))
+      onView(withId(R.id.profile_edit_name)).check(matches(withText("Ben")))
+      onView(withId(R.id.profile_edit_allow_download_container)).check(matches((isDisplayed())))
+      onView(withId(R.id.profile_delete_button)).check(matches((isDisplayed())))
+    }
+  }
+
+  @Test
+  fun testProfileEditActivity_startActivityWithUserProfile_clickRenameButton_checkOpensProfileRenameActivity() {
+    ActivityScenario.launch<ProfileEditActivity>(ProfileEditActivity.createProfileEditActivity(context, 1)).use {
+      onView(withId(R.id.profile_rename_button)).perform(click())
+      intended(hasComponent(ProfileRenameActivity::class.java.name))
+    }
+  }
+
+  @Test
+  fun testProfileEditActivity_startActivityWithUserProfile_clickResetPin_checkOpensProfileResetPinActivity() {
+    ActivityScenario.launch<ProfileEditActivity>(ProfileEditActivity.createProfileEditActivity(context, 1)).use {
+      onView(withId(R.id.profile_reset_button)).perform(click())
+      intended(hasComponent(ProfileResetPinActivity::class.java.name))
+    }
+  }
+
+  @Test
+  fun testProfileEditActivity_startActivityWithUserProfile_clickProfileDeletionButton_checkOpensDeletionDialog() {
+    ActivityScenario.launch<ProfileEditActivity>(ProfileEditActivity.createProfileEditActivity(context, 1)).use {
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      onView(withText(R.string.profile_edit_delete_dialog_message)).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testProfileEditActivity_startActivityWithUserProfile_clickProfileDeletionButton_clickDelete_checkReturnsToProfileListActivity() {
+    ActivityScenario.launch<ProfileEditActivity>(ProfileEditActivity.createProfileEditActivity(context, 1)).use {
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      onView(withText(R.string.profile_edit_delete_dialog_positive)).perform(click())
+      intended(hasComponent(ProfileListActivity::class.java.name))
+    }
+  }
+
+  @Test
+  fun testProfileEditActivity_startActivityWithUserHasDownloadAccess_checkSwitchIsChecked() {
+    profileManagementController.addProfile("James", "123", null, true, R.color.avatar_background_1, false)
```

Comment:
Please add inline comments for `addProfile ` parameters.

Reply:
Updated.